pandas/tests/frame/test_api.py

```diff
@@ -324,6 +324,22 @@ def test_attrs_deepcopy(self):
         assert result.attrs == df.attrs
         assert result.attrs["tags"] is not df.attrs["tags"]
 
+    def test_attrs_binary_operations(self):
+        # GH 51607
+        df_1 = DataFrame({"A": [2, 3]})
+        df_2 = DataFrame({"A": [-3, 9]})
+        attrs = {"info": "DataFrame"}
+        df_1.attrs = attrs
+        assert (df_1 + df_2).attrs == attrs
```

Comment:
Rather than doing this you can just use the `all_binary_operators` fixture from conftest.py (I think)

Reply:
I made the change.